feat: add Dockerfile support

setchy opened the pull request:
#657

setchy commented:
Opening a DRAFT PR early for feedback.  It's pretty rough, but appears to be functional locally when using a sample Dockerfile

prabhu commented on index.js:
```diff
@@ -3637,6 +3637,10 @@ export const createContainerSpecLikeBom = async (path, options) => {
   );
+  let dfFiles = getAllFiles(
+    path,
+    (options.multiProject ? "**/" : "") + "*Dockerfile"
```
Add another * at the end. For example, see cdxgen naming conventions under ci.

prabhu replied:
Containerfile is used by podman and other tools.

setchy replied:
Great feeedback.  Updated to support `Containerfile` and wider pattern expression.

prabhu commented on index.js:
```diff
@@ -3657,14 +3661,29 @@ export const createContainerSpecLikeBom = async (path, options) => {
+        for (const dfLine of dData.split("\n")) {
+          if (dfLine.includes("FROM")) {
+            imglist.push({
+              image: dfLine.split("FROM")[1].split("AS")[0].trim()
```
Do we need all the FROM statements or only the last one?

setchy replied:
We use a lot of mutli-stage Docker builds and I'd personally be interested in seeing each image used.

setchy commented:
A few updates

- updated logic to to skip processing any image statements which reuse a previous build stage name.
- created demo/test repository: https://github.com/setchy/cdxgen-dockerfile-demo
- Split out the multiple `SrcFile` support into a separate PR: https://github.com/CycloneDX/cdxgen/pull/696

While debugging, i noticed that `dependencies` were being found and logged, however they are not surviving through to the final output bom, which as I understand affects things like Dependency Graph within Dependency Track.  This needs some further investigation...

setchy commented:
I've split the `dependencies` enhancement into https://github.com/CycloneDX/cdxgen/issues/697

setchy commented:
re-raised as https://github.com/CycloneDX/cdxgen/pull/704
